relentless/potential/potential.py

```diff
@@ -34,28 +131,123 @@ def evaluate(self, pair):
                 params[p] = v
             elif v is not None:
                 raise TypeError('Parameter type unrecognized')
-
             if v is None:
                 raise ValueError('Parameter {} is not set for ({},{}).'.format(p,pair[0],pair[1]))
 
         return params
 
     def save(self, filename):
-        all_params = {}
+        """Saves the data from a coefficient matrix to a file.
+
+        Serializes each parameter value in order to track the internal values
+        of the :py:class:`Variable` object and distinguish it from 'scalar' values.
+
+        Parameters
+        ----------
+        filename : `str`
+            The name of the file to save data in.
+
+        """
+        data = {}
         for key in self:
-            all_params[str(key)] = self.evaluate(key)
+            all_params = {}
+            for param in self[key]:
+                var = self[key][param]
+                if isinstance(var, Variable):
+                    all_params[param] = {'type':'variable', 'value':var.value,
+                                         'const':var.const, 'low':var.low, 'high':var.high}
+                elif isinstance(var, (float,int)):
+                    all_params[param] = {'type':'scalar', 'value':var}
+            data[str(key)] = all_params
         with open(filename, 'w') as f:
-            json.dump(all_params, f, sort_keys=True, indent=4)
+            json.dump(data, f, sort_keys=True, indent=4)
 
     def load(self, filename):
-        raise NotImplementedError('Load is not implemented')
+        """Loads the data from a file into the coefficient matrix.
+
+        Parameters
+        ----------
+        filename : `str`
+            The name of the file from which to load data.
+
+        Raises
+        ------
+        KeyError
+            If the pairs in the data are not exactly the pairs in self
+        KeyError
+            If the params in the data are not exactly the params in self
+        TypeError
+            If the parameter value type in the data and in self do not match
+
+        """
+        with open(filename, 'r') as f:
+            data = json.load(f)
+        pairs = []
+        for key in data:
+            pairs.append(eval(key))
+        if sorted(self.pairs) != sorted(tuple(pairs)):
+            raise KeyError('The type pairs in the data are not exactly the pairs in self')
+        for key in data:
+            key_tup = eval(key)
+            if sorted(self.params) != sorted(data[key].keys()):
+                raise KeyError('The parameters in the data are not exactly the parameters in self')
+            for param in data[key]:
+                var_data = data[key][param]
+                var_self = self[key_tup][param]
```

Comment:
If you make the change I suggested above, then `key` will be the same as `key_tup`, and you should update this section. Also, I think you probably don't need a `var_self` since you also use `self[key_tup][param]` repeatedly below for setting, and I would prefer to just use the second.